lib/puppet_x/coi/jboss/provider/securitydomain.rb

```diff
@@ -1,15 +1,7 @@
 # A class for JBoss security domain provider
 module Puppet_X::Coi::Jboss::Provider::SecurityDomain
   def create
-    cmd = "/subsystem=security/security-domain=#{@resource[:name]}/authentication=classic:add(login-modules=[{code=>\"#{@resource[:code]}\",flag=>\"#{@resource[:codeflag]}\",module-options=>["
-    options = []
-    @resource[:moduleoptions].keys.sort.each do |key|
-      value = @resource[:moduleoptions][key]
-      val = value.to_s.gsub(/\n/, ' ').strip
-      options << '%s => "%s"' % [key, val]
-    end
-    cmd += options.join(',') + "]}])"
-    cmd = compilecmd(cmd)
+    cmd = compilecmd create_parametrized_cmd
```

Comment:
!resolve **Question:** I don't know that in this case is unneeded to invoke also: `subsystem=security/security-domain=testing/authentication=classic:add` before this for newer version of JBoss. It could fail on real JBoss